src/ameba/rule/lint/unused_local_variable_access.cr

```diff
@@ -0,0 +1,78 @@
+module Ameba::Rule::Lint
+  # A rule that disallows unused local variable access.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # a = 1
+  # b = "2"
+  # c = :3
+  #
+  # case method_call
+  # when Int32
+  #   a
+  # when String
+  #   b
+  # else
+  #   c
+  # end
+  #
+  # def hello(name)
+  #   if name.size < 10
+  #     name
+  #   end
+  #
+  #   name[...10]
+  # end
+  # ```
+  #
+  # And these are considered valid:
+  #
+  # ```
+  # a = 1
+  # b = "2"
+  # c = :3
+  #
+  # d = case method_call
+  #     when Int32
+  #       a
+  #     when String
+  #       b
+  #     else
+  #       c
+  #     end
+  #
+  # def hello(name)
+  #   if name.size < 10
+  #     return name
+  #   end
+  #
+  #   name[...10]
+  # end
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Lint/UnusedLocalVariableAccess:
+  #   Enabled: true
+  # ```
+  class UnusedLocalVariableAccess < Base
+    properties do
+      since_version "1.7.0"
+      description "Disallows unused access to local variables"
+    end
+
+    MSG = "Value from local variable access is unused"
+
+    def test(source : Source)
+      AST::ImplicitReturnVisitor.new(self, source)
+    end
+
+    def test(source, node : Crystal::Var, node_is_used : Bool)
+      return if node_is_used || node.name.in?("skip_file", "self")
```

Comment:
Why `skip_file` variable name has a special treatment?

Reply:
For some reason the parser treats it as a var even though it's a call. This should probably be updated to only skip that (and probably `debug` if that's also an issue) when inside macros. I have another branch waiting to improve macro support in the implicit return visitor, I can remove this until then.

https://crystaldoc.info/github/crystal-lang/crystal/1.15.0/Crystal/Macros.html#skip_file:Nop-instance-method

Reply:
I'd prefer doing so. Btw, if possible it would be nice to merge the refactors re: macro support first.